Source/monster.cpp

```diff
@@ -3970,15 +3970,12 @@ void StartMonsterDeath(Monster &monster, int pnum, bool sendmsg)
 	MonsterDeath(monster, pnum, md, sendmsg);
 }
 
-void M_StartKill(int monsterId, int pnum)
+void M_StartKill(Monster &monster, int pnum)
 {
-	assert(monsterId >= 0 && monsterId < MaxMonsters);
-	Monster &monster = Monsters[monsterId];
-
 	if (pnum == MyPlayerId) {
-		delta_kill_monster(monsterId, monster.position.tile, *MyPlayer);
-		if (monsterId != pnum) {
-			NetSendCmdLocParam1(false, CMD_MONSTDEATH, monster.position.tile, monsterId);
+		delta_kill_monster(monster.getId(), monster.position.tile, *MyPlayer);
+		if (monster.getId() != pnum) {

```

Comment:
This introduces the following warning:

```
warning: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'int' [-Wsign-compare]
```

But I am not sure how to *properly* solve it. Monster::getId returns a size_t but the playerId are int

Reply:
`static_cast<size_t>(pnum)`

Reply:
But that's just hidding the issue, no?

Reply:
It makes it explicit that the comparison is intended

Reply:
Thanks! I updated it